examples/gno.land/r/sys/validators/validators.gno

```diff
@@ -1,4 +1,34 @@
-// This package is used to manage the validator set.
+// Package validators is used to manage the validator set.
 package validators
 
-// write specs.
+import (
+	"std"
+
+	"gno.land/p/gov/proposal"
+)
+
+// Change represents a change in the validator set.
+type Change struct {
+	Address std.Address
+	Power   int
+}
+
+// NewProposalExecutor creates a new executor that wraps a changes closure
```

Comment:
I've been thinking a lot about how we can integrate something like this in the system I put in place in https://github.com/gnolang/gno/pull/2130, where I essentially defined an API for a validator set protocol

If I'm understanding your direction in this PR, we want the proposal to define a closure that will be executed, and DAO members vote on the proposal. If by some mechanism the proposal is accepted, the closure is executed. 

In the case of the #2130 system, the closure would simply call `r/sys/vals` methods `AddValidator` or `RemoveValidator`, and the underlying PoC implementation will verify the action (most probably by verifying govdao executed the action, indicating the proposal is valid and can't be spoofed)

Reply:
The PR can be patched to continue using `std.Emit` (for external subscribes), while implementing a simple logic using a global slice of changes to apply.

`gnoland` can call an unexported function that "reads and resets" the changes.

I can make the patch if needed.